pkg/core/interop/storage/find.go

```diff
@@ -58,7 +61,7 @@ func (s *Iterator) Value() stackitem.Item {
 	}
 	key := s.curr.Key
 	if s.opts&FindRemovePrefix == 0 {
-		key = append(s.prefix, key...)
+		key = append(slice.Copy(s.prefix), key...)
```

Comment:
Refs. #3310 as well, perfect case for https://pkg.go.dev/slices#Concat (likely we have a number of similar ones as well).